packages/typespec-azure-resource-manager/src/rules/utils.ts

```diff
@@ -93,6 +93,14 @@ export function isInternalTypeSpec(
   );
 }
 
+export function isLegacyTypeSpec(
```

Comment:
I am not sure where this is used, but I would not expect Service TypeSpec to use `.Legacy` in their namespaces.

I think we should instead add a new linting rule that throws if folks use customAzureResource, and think about having a new 'legacy' ruleset that does not include this rule.  But I don't think the new ruleset needs to be in this PR

Reply:
removed this check